src/llmcompressor/modifiers/quantization/gptq/utils/gptq_wrapper.py

```diff
@@ -98,6 +99,40 @@ def compress(
             W = W.t()
         W = W.float()
 
+        tick = time.time()
+
+        # if activation ordering is enabled, permute the weight columns
+        # in order of greatest hessian values. Columns are unpermuted after
+        # quantization is finished
+        actorder = False
+        if hasattr(self.layer, "quantization_scheme"):
+            quant_scheme = self.layer.quantization_scheme
+            quant_weights = quant_scheme.weights
+            if quant_weights is not None:
+                actorder = quant_weights.actorder
+                if actorder:
+                    # use hessian to create a permutation of weights
+                    perm = torch.argsort(torch.diag(H), descending=True)
+
+                    # permute weight and hessian
+                    W = W[:, perm]
+                    H = H[perm][:, perm]
+
+            # fetch latest correct scale and ZP relevant for any changes
+            from compressed_tensors.quantization import update_layer_weight_quant_params
+
+            # TODO: experiment with updating before each block
+            update_layer_weight_quant_params(self.layer, weight=W, reset_obs=True)
+            scale = self.layer.weight_scale.data
+            zero_point = self.layer.weight_zero_point.data
```

Comment:
Since the weight value changes between block iterations, the quantization parameters may change. Updating after each weight value change would technically be more faithful to the true values